dissect/target/plugins/apps/texteditor/windowsnotepad.py

```diff
@@ -0,0 +1,184 @@
+    uleb128                     fsize1;
+    uleb128                     fsize2;
+    char                        header_end[2];    // \x01\x00
+
+    // Data can be stored in two says:
+    //  1. A single, contiguous block of data that holds all the data
+    //     In this case, the header is included in the single CRC32 checksum present at the end of the block
+    //  2. Multiple blocks of data that, when combined, hold all the data
+    //     In this case, the header has a separate CRC32 value stored at the end of the header
+    // The following bitmask operations basically check whether fsize1 is nonzero (boolean check) and depending
+    // on the outcome, parse 0 or 1 (so basically, parse or not parse) structs.
+    header_crc                  header_crc[((fsize1 | -fsize1) >> 31) ^ 1]; // Optional, only if fsize1 == 0
+    single_block_entry          single_block_entry[((fsize1 | (~fsize1 + 1)) >> 31) & 1];  // Optional, only if fsize1 > 0  # noqa: E501
+
+
+    multi_block_entry           multi_block_entries[EOF];  // Optional. If a single_block_entry is present
+                                                           // this will already be at EOF, so it won't do anything.
+                                                           // Otherwise, it will parse the individual blocks.
+};
+"""
+TextEditorTabRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+    "texteditor/windowsnotepad/tab", GENERIC_TAB_CONTENTS_RECORD_FIELDS
+)
+
+c_windowstab = cstruct.cstruct()
+c_windowstab.load(c_def)
+
+
+def _calc_crc32(data: bytes) -> bytes:
+    """Perform a CRC32 checksum on the data and return it as bytes"""
+    return zlib.crc32(data).to_bytes(length=4, byteorder="big")
+
+
+class WindowsNotepadPlugin(TexteditorPlugin):
+    """Windows notepad tab content plugin."""
+
+    __namespace__ = "windowsnotepad"
+
+    GLOB = "AppData/Local/Packages/Microsoft.WindowsNotepad_*/LocalState/TabState/*.bin"
+
+    def __init__(self, target):
+        super().__init__(target)
+        self.users_tabs = []
+
+        for user_details in self.target.user_details.all_with_home():
+            for tab_file in user_details.home_path.glob(self.GLOB):
+                if tab_file.name.endswith(".1.bin") or tab_file.name.endswith(".0.bin"):
+                    continue
+
+                self.users_tabs.append(tab_file)
+
+    def check_compatible(self) -> None:
+        if not self.users_tabs:
+            raise UnsupportedPluginError("No Windows Notepad temporary tab files found")
+
+    def _process_tab_file(self, file: TargetPath) -> TextEditorTabRecord:
+        """
+        Function that parses a binary tab file and reconstructs the contents.
+
+        Args:
+            file: The binary file on disk that needs to be parsed.
+
+        Returns:
+            A TextEditorTabRecord containing information that is in the tab.
+        """
+        fh: BinaryIO = file.open(mode="rb")
```

Comment:
```suggestion
        fh: BinaryIO = file.open("rb")
```

It doesn't matter much in Dissect now, but can you use a context manager?

Reply:
Yes, refactored it to use a context manager.